lib/note-detail.jsx

```diff
@@ -1,62 +1,164 @@
-import React, { PropTypes } from 'react';
-import marked from 'marked';
+import React, { Component, PropTypes } from 'react';
 import Textarea from 'react-textarea-autosize';
-import { noop, get, debounce } from 'lodash';
 import analytics from './analytics';
+import marked from 'marked';
+import { get, debounce, invoke } from 'lodash';
 import { viewExternalUrl } from './utils/url-utils';
 
-const uninitializedNoteEditor = { focus: noop };
 const saveDelay = 2000;
 
-export default React.createClass( {
+const isValidNote = note => note && note.id;
+const prependTab = l => `\t${ l }`;
+const removeLeadingTab = l => l && l.length && l[0] === '\t' ? l.substring( 1 ) : l;
 
-	propTypes: {
+export default class NoteDetail extends Component {
+	static propTypes = {
 		note: PropTypes.object,
 		previewingMarkdown: PropTypes.bool,
 		fontSize: PropTypes.number,
 		onChangeContent: PropTypes.func.isRequired
-	},
+	};
+
+	constructor( props ) {
+		super( props );
 
-	componentWillMount: function() {
+		this.storeNoteEditor = r => this.noteEditor = r;
 		this.queueNoteSave = debounce( this.saveNote, saveDelay );
-		this.noteEditor = uninitializedNoteEditor;
-	},
+	}
 
-	componentDidMount: function() {
+	componentDidMount = () => {
```

Comment:
nope, you only need to bind class methods to an instance when you use them "detached" from the a reference to the instance, ie.
```js
const obj = {
  abc: 2,

  boo() {
    console.log(this.abc)
  }
}

obj.boo() // logs `2`
// this is akin to how react internally calls lifecycle methods

const boo = this.boo

boo() // fails because `this` is not defined
// because you're not calling this method on any object (and the function isn't previously bound to any context), so there's no context to pass to the function
// this is akin to how event handler end up being called by react, because by doing `<div onClick={this.boo} />` you're passing a reference to the function boo, like in `const boo = this.boo` above
```

long story short, do bind event handlers, don't bind lifecycle methods